pkg/kritis/metadata/containeranalysis/containeranalysis.go

```diff
@@ -128,29 +129,53 @@ func (c Client) fetchAttestationOccurrence(containerImage string, kind string, a
 		return nil, fmt.Errorf("%s is not a valid image hosted in GCR", containerImage)
 	}
 
+	noteName := fmt.Sprintf("%s/notes/%s", auth.Spec.NoteReference, auth.Name)
 	req := &grafeas.ListNoteOccurrencesRequest{
-		Name: auth.Spec.NoteReference,
-		// TODO: re-add `kind` clause in filter
-		//       (currently not supported as per https://github.com/grafeas/kritis/issues/401#issuecomment-538947608)
+		Name: noteName,
 		// Example:
 		// 		Filter:  fmt.Sprintf("resourceUrl=%q AND kind=%q", util.GetResourceURL(containerImage), kind),
 		Filter:   fmt.Sprintf("resourceUrl=%q", util.GetResourceURL(containerImage)),
 		PageSize: constants.PageSize,
 	}
 
-	it := c.client.ListNoteOccurrences(c.ctx, req)
+	return c.listNoteOccurrences(req)
+}
+
+func (c Client) listNoteOccurrences(req *grafeas.ListNoteOccurrencesRequest) ([]*grafeas.Occurrence, error) {
```

Comment:
I don't really understand how this function works:
1. Where is pagination handled?
2. What is the purpose of fetching Occurrences only when a clock tick happens?

Reply:
Because listing note occurrences gives results with a staleness bound, the goal of this function is to give the opportunity for the newly created attestations to be retrieved from `containeranalysis`. To answer your specific questions:

1. The client doesn't expose pagination logic, I suspect because it's handled internally already: [reference](https://godoc.org/cloud.google.com/go/containeranalysis/apiv1beta1#GrafeasV1Beta1Client.ListNoteOccurrences). I filed #419 to verify this is actually the case.

2. This is to ensure we don't send too many `ListNoteOccurrences` requests in a given time unit. Because of the staleness bound, there's no point to actively be checking if occurrences have propagated through the system after receiving none.

Question for you @nenaddedic: in BinAuthz, do you expect to have multiple occurrences per attestor?

Reply:
Multiple occurrences per image per attestor would be uncommon, but possible.  (I assume you meant "per image per attestor".)

Polling for attestations has its advantages, but I think they're outweighed by disadvantages:
1. The common "no attestations" rejection now necessarily takes 20 seconds to reach.
2. Code clarity.  It is not clear why keep on polling only when "no occurrences" is returned.  Timeouts (L154) should arguably be handled in this way as well.  Maybe also unverifiable attestations.

I suggest removing the polling here and changing the failing integration test to retry for 20 seconds.  (Unless that turns out to be more work than you can afford now.)

Reply:
We discussed this in more detail yesterday: @nenaddedic made good points, and we agreed on the approach. I added the comments to reflect our current understanding.